examples/02_scope/src/App.tsx

```diff
@@ -1,43 +1,87 @@
 import { atom, useAtom } from 'jotai';
 import { ScopeProvider } from 'jotai-scope';
+import { useHydrateAtoms } from 'jotai/react/utils';
+import { PropsWithChildren } from 'react';
 
-const countAtom = atom(0);
-const anotherCountAtom = atom(0);
+const primitiveAtom = atom(0);
+const scopedAtom = atom(1);
+const derivedAtom1 = atom((get) => get(scopedAtom) + 1);
+const derivedAtom2 = atom((get) => get(primitiveAtom) + get(scopedAtom) + 3);
 
 const Counter = () => {
-  const [count, setCount] = useAtom(countAtom);
-  const [anotherCount, setAnotherCount] = useAtom(anotherCountAtom);
+  const [p, setP] = useAtom(primitiveAtom);
+  const [s, setS] = useAtom(scopedAtom);
+  const [d1] = useAtom(derivedAtom1);
+  const [d2] = useAtom(derivedAtom2);
   return (
     <>
       <div>
-        <span>count: {count}</span>
-        <button type="button" onClick={() => setCount((c) => c + 1)}>
+        <span>p: {p}</span>
+        <button type="button" onClick={() => setP((x) => x + 1)}>
           increment
         </button>
       </div>
       <div>
-        <span>another count: {anotherCount}</span>
-        <button type="button" onClick={() => setAnotherCount((c) => c + 1)}>
+        <span>s: {s}</span>
+        <button type="button" onClick={() => setS((x) => x + 1)}>
           increment
         </button>
       </div>
+      <div>
+        <span>d1: {d1}</span>
+      </div>
+      <div>
+        <span>d2: {d2}</span>
+      </div>
     </>
   );
 };
 
-const App = () => {
+function ScopeProviderWithInitializer({
+  atomValues,
+  children,
+}: PropsWithChildren<{
+  atomValues: Parameters<typeof useHydrateAtoms>[0];
+}>) {
+  const atoms = Array.from(atomValues, ([anAtom]) => anAtom);
+  return (
+    <ScopeProvider atoms={atoms}>
+      <AtomsHydrator atomValues={atomValues}>{children}</AtomsHydrator>
+    </ScopeProvider>
+  );
+}
+
+function AtomsHydrator({
+  atomValues,
+  children,
+}: PropsWithChildren<{
+  atomValues: Parameters<typeof useHydrateAtoms>[0];
+}>) {
+  useHydrateAtoms(atomValues);
+  return <>{children}</>;
+}
+
+export default function App() {
   return (
     <div>
+      <h1>Global</h1>
+      <Counter />
       <h1>First Provider</h1>
-      <ScopeProvider atoms={[anotherCountAtom]}>
+      <ScopeProvider atoms={[scopedAtom]}>
         <Counter />
       </ScopeProvider>
       <h1>Second Provider</h1>
-      <ScopeProvider atoms={[anotherCountAtom]}>
+      <ScopeProvider atoms={[scopedAtom]}>
         <Counter />
       </ScopeProvider>
+      <h1>Provider with initial value 10</h1>
+      <ScopeProviderWithInitializer
+        atomValues={
+          [[scopedAtom, 10]] as unknown as Parameters<typeof useHydrateAtoms>[0]
```

Comment:
Is it possible to export more types from jotai core repo? I see the original parameter signature uses `infer` trick, not sure how to make `useHydrateAtom`'s wrapper's type signature compatible